File: UI/travle_core/lib/src/models/revenue_report.dart
import 'package:json_annotation/json_annotation.dart';

import 'revenue_group_row.dart';

part 'revenue_report.g.dart';

/// The revenue-by-category/region report (mirrors the backend `RevenueReport`): two
/// single-GroupBy breakdowns of the same captured payments plus the reconciling
/// grand totals. Revenue is attributed to each tour's primary destination.
@JsonSerializable()
class RevenueReport {
  RevenueReport({
    required this.byCategory,
    required this.byRegion,
    required this.totalGross,
    required this.totalRefunded,
    required this.totalNet,
    required this.currency,
    this.fromDate,
    this.toDate,
  });

  final List<RevenueGroupRow> byCategory;
  final List<RevenueGroupRow> byRegion;
  final double totalGross;
  final double totalRefunded;
  final double totalNet;
  final String currency;
  final DateTime? fromDate;
  final DateTime? toDate;

  factory RevenueReport.fromJson(Map<String, dynamic> json) =>
      _$RevenueReportFromJson(json);

  Map<String, dynamic> toJson() => _$RevenueReportToJson(this);
}

File: Backend/Travle.Model/Responses/RevenueReport.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// The "revenue by category / region" report. Two single-GroupBy breakdowns of the same captured
    /// payments over the period — one keyed by the primary destination's category, one by its region —
    /// plus the reconciling grand totals. Used by both the JSON preview endpoint and the PDF endpoint.
    /// </summary>
    public class RevenueReport
    {
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }

        public List<RevenueGroupRow> ByCategory { get; set; } = new();
        public List<RevenueGroupRow> ByRegion { get; set; } = new();

        public decimal TotalGross { get; set; }
        public decimal TotalRefunded { get; set; }
        public decimal TotalNet { get; set; }

        public string Currency { get; set; } = "bam";
    }
}
